aws/resource_aws_msk_cluster.go

```diff
@@ -587,6 +586,41 @@ func resourceAwsMskClusterUpdate(d *schema.ResourceData, meta interface{}) error
 		}
 	}
 
+	if d.HasChange("kafka_version") {
+		var input *kafka.UpdateClusterKafkaVersionInput
```

Comment:
Can the kafka version be downgraded? If not, it would be good to check that the new version is higher than the old one.

Reply:
MSK cannot be downgraded AFAIK

Reply:
Yes, Kafka version can't be downgraded. But the option to upgrade may also not be available between certain versions. Does it make more sense to leave it up to the API to throw an error incase an upgrade is not possible?

Reply:
Hi @arafsheikh, continuing the review here 👋  Given the version can't be downgraded it could make sense to keep the `ForceNew` behavior but through a CustomizeDiff function like seen in the `sagemaker_notebook_instance` [resource definition](https://github.com/hashicorp/terraform-provider-aws/blob/0eae9a30ac2565cce682a083f3c5ae55c0477e5a/aws/resource_aws_sagemaker_notebook_instance.go#L28) or something along the lines of:
```go
CustomizeDiff: customdiff.Sequence(
    customdiff.ForceNewIfChange("kafka_version", func(_ context.Context, old, new, meta interface{}) bool {
        return new.(string) < old.(string)
    }),
),
```
This way the resource would go through the necessary destroy/create flow without additional intervention.

Reply:
For the case of not being able to upgrade, leaving it up to the API makes sense as it may become difficult to maintain the versions supported upstream 👍

Reply:
Done. Added `TestAccAWSMskCluster_KafkaVersionDowngrade` as well.